### ISSUE TITLE
Docksal virtualbox update caused higher CPU and energy consumtion 

### ISSUE THREAD

**Issue by awm086** [docksal/docksal#1032]
Description: 
I am not sure if this is an issue with Docksal, after updating to Docksal "1.85.1" I am seeing noticeably higher CPU/Enegry consumption coming from VboxHeadless even in idle state. 

Apologies if this does not belong here but I am only opening this issue just to check if this is a Virtualbox issue or perhaps a the way docksal is provisioning/configuring the vm.  This could be related 
1. Virtual box issue:  https://www.virtualbox.org/ticket/18089?cversion=0&cnum_hist=16 
2. Docksal utilizing more resources:  https://github.com/docksal/docksal/pull/999 : does this mean higher CPU/Higher energy usage? is it possible to configure it via the virtual box UI? 

- https://i.imgur.com/FWdbsdEl.png (note the idle wake ups in the screen shot is 1000+ which is too hight compared to others). 
- process appear to be using high cpu in idle state: https://i.imgur.com/jf6VfCi.png

Sysinfo:
```
███  OS
Darwin Mac OS X 10.12.6
Darwin MacBook-Pro.local 16.7.0 Darwin Kernel Version 16.7.0: Mon Nov 13 21:56:25 PST 2017; root:xnu-3789.72.11~1/RELEASE_X86_64 x86_64

███  ENVIRONMENT
MODE : VirtualBox VM
DOCKER_HOST : tcp://192.168.64.100:2376
DOCKSAL_NFS_PATH : /Users/ali/docksal-sites
NFS EXPORTS:

# <ds-nfs docksal
/Users/ali/docksal-sites 192.168.64.1 192.168.64.100 -alldirs -maproot=0:0
# ds-nfs>

███  FIN
fin version: 1.85.1

███  DOCKER COMPOSE
EXPECTED VERSION: 1.23.2
docker-compose version 1.23.2, build 1110ad01
docker-py version: 3.6.0
CPython version: 3.6.6
OpenSSL version: OpenSSL 1.1.0h  27 Mar 2018

███  DOCKER
EXPECTED VERSION: 18.09.2

Client: Docker Engine - Community
 Version:           18.09.2
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        6247962
 Built:             Sun Feb 10 04:12:39 2019
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.2
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.6
  Git commit:       6247962
  Built:            Sun Feb 10 04:20:28 2019
  OS/Arch:          linux/amd64
  Experimental:     false

███  DOCKER MACHINE
EXPECTED VERSION: 0.16.1
docker-machine version 0.16.1, build cce350d7

NAME      ACTIVE   DRIVER       STATE     URL                         SWARM   DOCKER     ERRORS
docksal   *        virtualbox   Running   tcp://192.168.64.100:2376           v18.09.2   

███  DOCKSAL: PROJECTS
project             STATUS              virtual host        project root

███  DOCKSAL: VIRTUAL HOSTS
No virtual hosts found

███  DOCKSAL: DNS
Successfully requested http://dns-test.docksal

███  DOCKER: RUNNING CONTAINERS
CONTAINER ID        IMAGE                     COMMAND                  CREATED              STATUS                        PORTS                                                    NAMES
f55e5329b92c        docksal/ssh-agent:1.2     "docker-entrypoint.s…"   About a minute ago   Up About a minute (healthy)                                                            docksal-ssh-agent
71d6b8d94e07        docksal/dns:1.1           "docker-entrypoint.s…"   About a minute ago   Up About a minute (healthy)   192.168.64.100:53->53/udp                                docksal-dns
0d5b485b23fc        docksal/vhost-proxy:1.5   "docker-entrypoint.s…"   About a minute ago   Up About a minute (healthy)   192.168.64.100:80->80/tcp, 192.168.64.100:443->443/tcp   docksal-vhost-proxy

███  DOCKER: NETWORKS
NETWORK ID          NAME                      DRIVER              SCOPE
****REDACTED*** 


███  VIRTUALBOX
EXPECTED VERSION: 5.2.26
5.2.26r128414

███  DOCKSAL MOUNTS
....

███  HDD Usage
Filesystem                Size      Used Available Use% Mounted on
/dev/sda1                46.1G      5.8G     37.9G  13% /mnt/sda1
```

I will try to update virtual box to version 6.* to see if it's better.

**Comment by lmakarov** [comment]
@awm086 please post the output from `fin sysinfo` so we have a record of your current configuration.

The default CPU count in VirtualBox is now set to 2 (#999). The VM gets more hardware resources from the host, so seeing a higher host CPU Usage/Energy impact is expected.
At the same time, virtualizing CPUs has its own overhead for hypervisors, so a single CPU may perform better than 2 or more CPUs in certain cases. It all depends on the kind of workloads you'd be running the most. We've debated this change internally, but decided to give it a try.

If you see a higher CPU usage while the VM/stack is idle, that's not good. The idle CPU usage is likely related to service-vhost-proxy/#44. I wonder if that has now worsened with 2 CPUs in VirtualBox.

Try removing (or setting to "1") the following line in `/usr/local/bin/fin`, recreate the VM and compare the results:
https://github.com/docksal/docksal/blob/4ded52a756c2edff6af222f427a067c712b1ebce/bin/fin#L2336

Consider running some tests mentioned in #999 to compare the stack performance when changing the CPU count.

CC @achekulaev

**Comment by awm086** [comment]
@lmakarov  updated the original comment.

**Comment by awm086** [comment]
@lmakarov does limiting the docksal-vhost-prox resource usage have a negative impact on the stack. I tried setting the cpu usage of the container to 5% and it seems to function fine with much reduced cpu footprint for the entire stack.
 I just added  `--cpu=".05"` to 
`https://github.com/docksal/docksal/blob/develop/bin/fin#L3723`

**Comment by lmakarov** [comment]
> I tried setting the cpu usage of the container to 5% and it seems to function fine

@awm086 thats a quick, dirty and brilliant workaround! 😁 
This will likely slow down nginx configuration updates inside vhost-proxy, but that may not be a problem a all for local development needs. I'm going to try this on my instance and see how it goes.

**Comment by lmakarov** [comment]
Should be `--cpus=".05"` (not `--cpu=".05"`). I updated your comment.

**Comment by awm086** [comment]
@lmakarov  sorry `	docker run -d --name docksal-vhost-proxy  --cpus=".05"  --label "io.docksal.group=system" --restart=unless-stopped \
`

see https://docs.docker.com/config/containers/resource_constraints.

And I have been using it with 1 core and `--cpu=".05"` and I feel performance impact has so far been negligible in terms of normal drupal use. But battery and cpu consumption back to "normal" (how it used to be) level.

**Comment by lmakarov** [comment]
This has been addressed in docksal/service-vhost-proxy#48 and released as a hotfix in docksal/vhost-proxy [v1.5.1](https://github.com/docksal/service-vhost-proxy/releases/tag/v1.5.1).

Run `fin update` to get the fix.

Please use docksal/service-vhost-proxy#44 as the main issue for any further followups.
